### PR TITLE
Dependency node cleanup

### PR DESCRIPTION
**Pull request by drewnoakes**
Small changes pulled out of larger work on the dependencies node.

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/OnceInitializedOnceDisposedUnderLockAsync.cs [hunk 1]
```diff
@@ -112,29 +114,29 @@ private async Task<T> ExecuteUnderLockCoreAsync<T>(Func<CancellationToken, Task<
...
+                if (!TryTranslateException(ex, cancellationToken, DisposalToken))
+                    throw;
             }
 
             throw Assumes.NotReachable();
         }
 
         private async Task ExecuteUnderLockCoreAsync(Func<CancellationToken, Task> action, CancellationToken cancellationToken = default)
         {
+            Requires.NotNull(action, nameof(action));
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/OnceInitializedOnceDisposedUnderLockAsync.cs]: Why?

Comment (drewnoakes) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/OnceInitializedOnceDisposedUnderLockAsync.cs]: Per 890e3b9fafad516597d102454c32e25cfa79c806, makes it consistent with the other methods in this class. Either they should all be removed, or all added. I am less bullish than you on the idea that NRT means we don't need runtime checks, for now at least.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/Tree/Dependencies/Snapshot/AggregateDependenciesSnapshotProvider.cs [hunk 2]
```diff
@@ -36,10 +36,7 @@ internal class AggregateDependenciesSnapshotProvider : IAggregateDependenciesSna
         /// <inheritdoc />
         public void RegisterSnapshotProvider(IDependenciesSnapshotProvider snapshotProvider)
         {
-            if (snapshotProvider == null)
-            {
-                return;
-            }
+            Requires.NotNull(snapshotProvider, nameof(snapshotProvider));
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/Tree/Dependencies/Snapshot/AggregateDependenciesSnapshotProvider.cs]: @davidwengier I assume you're referring to this? If so then it's fine, no caller passes `null`.

Comment (davidwengier) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/Tree/Dependencies/Snapshot/AggregateDependenciesSnapshotProvider.cs]: Yep, that's the one.

**Pull request merged**